soc/silabs/silabs_sim3/sim3u/Kconfig

```diff
@@ -8,3 +8,4 @@ config SOC_SERIES_SIM3U
 	select CPU_CORTEX_M3
 	select CPU_CORTEX_M_HAS_SYSTICK
 	select HAS_SILABS_SI32
+	select SOC_EARLY_INIT_HOOK
```

Comment:
This makes sense since you're modifying `soc/silabs/silabs_sim3/sim3u/soc.c`. However, you're also modifying `soc/silabs/common/soc.c`, and I don't see you adding `select SOC_EARLY_INIT_HOOK` to any other Silabs-related Kconfig file. @asmellby FYI

Reply:
Looks like the missing places are in `soc/silabs/silabs_s[012]/Kconfig`